### DISCUSSION TITLE
Can I enable hot reloading or anything to not have to logout and back in to see changes to my plugin?

### DISCUSSION THREAD

**Discussion by elcarn23** [cockpit-project/cockpit#16261] [Q&A] [answered]
Hello  I'm working on a plugin for cockpit and I've noticed in order to see changes in the plugin I have to logout and then log back in.  I recently found this repo and the cockpit starter project that uses webpack and react and all those goodies but even with that I when updating components webpack rebuilds just fine and I noticed it doesn't hot reload and even with a hard refresh I can only get the plugin to update once with the changes and after that I have to logout and then back in again for it to pick up the changes.  It's a little frustrating.  I'm not sure if I'm just doing something wrong but I would think the starter-project wouldn't have that problem.  Any help and guidance would be appreciated.

**Comment by marusak** [comment]
Interesting. What kind of changes are you doing that need log out and back in? I don't need to do it and just simple reload always works. Do you symlink your plugin into `~/.local/share/cockpit/`?

Auto refresh is not available in our development process.

**Comment by elcarn23** [comment]
No the original documentation that I was following when setting up cockpit with my plugin said to symlink the plugin into ~/usr/share/cockpit/  I've created the symlink in ~/.local/share/cockpit and even after that removed the one from /usr/share/cockpit and now it's not even showing any changes I have to remove the link refresh cockpit to show the plugin is gone then add the link back in and refresh to get the plugin back and it has the changes.

**Reply by elcarn23** [reply]
Also this is me developing this on Fedora 34 running the Cockpit Service in Chrome with adminstrative access.  I've also started from scratch with the starter kit again.  Following each step and doing the watch portion so webpack rebuilds everything into the dist directory and I don't see any updates to my plugin when refreshing.  Even more interesting is after moving to the ~./local/share/cockpit location for the symlink I can't even log out and back in for changes.  I have to remove and re add the symlink with refreshing the page in between for it to pick up the new changes.

**Reply by elcarn23** [reply]
For the record when testing it out in Firefox I'm not running into these issues.  Seems to be an issue with Chrome.  Maybe it's caching the first time it loads or something and not looking for the new files after that.

**Comment by elcarn23** [comment, marked as answer]
As noted in my reply to myself.  Firefox was the answer to get the updates from the webpack builds.
